src/services/donationService.ts

```diff
@@ -132,3 +134,68 @@ export const isTokenAcceptableForProject = async (inputData: {
     return false;
   }
 };
+
+const toFixNumber = (input: number, digits: number): number => {
+  return convertExponentialNumber(Number(input.toFixed(digits)));
+};
+
+export const updateOldGivDonationsPrice = async () => {
+  const donations = await Donation.findXdaiGivDonationsWithoutPrice();
+  logger.debug('updateOldGivDonationPrice donations count', donations.length);
+  for (const donation of donations) {
+    logger.debug(
+      'updateOldGivDonationPrice() updating accurate price, donationId',
+      donation.id,
+    );
+    try {
+      const givHistoricPrices = await fetchGivHistoricPrice(
+        donation.transactionId,
+        donation.transactionNetworkId,
+      );
+      logger.debug('Update donation usd price ', {
+        donationId: donation.id,
+        ...givHistoricPrices,
+        valueEth: toFixNumber(
+          donation.amount * givHistoricPrices.givPriceInEth,
+          6,
+        ),
+      });
+      donation.priceEth = toFixNumber(givHistoricPrices.ethPriceInUsd, 6);
+      donation.priceUsd = toFixNumber(givHistoricPrices.givPriceInUsd, 3);
+      donation.valueUsd = toFixNumber(
+        donation.amount * givHistoricPrices.givPriceInUsd,
+        3,
+      );
+      donation.valueEth = toFixNumber(
+        donation.amount * givHistoricPrices.givPriceInEth,
+        6,
+      );
+      await donation.save();
+      await updateTotalDonationsOfProject(donation.projectId);
+    } catch (e) {
+      logger.error('Update GIV donation valueUsd error', e.message);
+    }
+  }
+};
+
+export const updateOldStableCoinDonationsPrice = async () => {
+  const donations = await Donation.findStableCoinDonationsWithoutPrice();
+  logger.debug(
+    'updateOldStableCoinDonationPrice donations count',
+    donations.length,
+  );
+  for (const donation of donations) {
+    logger.debug(
+      'updateOldStableCoinDonationPrice() updating accurate price, donationId',
+      donation.id,
+    );
+    try {
+      donation.priceUsd = 1;
```

Comment:
In this case we are setting values to 1, and ignoring any variance, for instance if in that moment the value was 0.98?

Reply:
You know as you  said it's hard to get accurate price, and it's not necessary to do that, in trace.giveth.io we assume every stable coin is 1$ as well ( I didn't include **UST** here :))) )